server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputerTests.java

```diff
@@ -366,6 +372,132 @@ public void testNoDataNodes() {
         assertDesiredAssignments(desiredBalance, Map.of());
     }
 
+    public void testShouldDesiredBalanceConverges() {
```

Comment:
Currently this has passed after 1000 iterations.
I suspect the issue with `testDelayWithALargeAmountOfShards` might be around `INDEX_DELAYED_NODE_LEFT_TIMEOUT_SETTING`